### PR TITLE
Fix for issue #47216

### PR DESCRIPTION
**Pull request by shivaditya-meduri**
FIX TypeError if set the weights to the current weights via `set_weights, more details in #47216.
This new line improves the versatility of the set_weights() function. Now we can use .get_weights() or .weights of any existing layer to the set_weights() of a new layer to copy the existing weights to the new layer. @chenmoneygithub  I fixed the issue mentioned in #47255

### CODE REVIEW

File: tensorflow/python/keras/engine/base_layer.py [hunk 1]
```diff
@@ -1868,6 +1868,8 @@ def set_weights(self, weights):
           'expecting %s weights. Provided weights: %s...' %
           (self.name, len(weights), expected_num_weights, str(weights)[:50]))
 
+    weights = [np.array(item) for item in weights]
```
Comment (qlzh727) [inline tensorflow/python/keras/engine/base_layer.py:1871]: Any reason to force convert the into into np.array (which might have non-trival performance overhead)? Also it might hit issue when assigning sparse/ragged variable. 

If this is just trying to avoid ppl from using layer_a.set_weights(layer_b.weights()), shouldn't we just validate the weights here and see if it contains any tf.variable types?

**Review by qlzh727 — CHANGES_REQUESTED** [review]

File: tensorflow/python/keras/engine/base_layer.py (hunk 1 above)
Comment (shivaditya-meduri) [reply tensorflow/python/keras/engine/base_layer.py:1871]: That's a good point, I will make the changes and commit again. Thanks for your time.

Comment (shivaditya-meduri) [reply tensorflow/python/keras/engine/base_layer.py:1871]: Please don't close this PR, I will commit to this PR again.

Comment (gbaned) [reply tensorflow/python/keras/engine/base_layer.py:1871]: @shivaditya-meduri Any update on this PR? Please. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by shivaditya-meduri** [comment]
on it

**Comment by gbaned** [comment]
@shivaditya-meduri  Any update on this PR? Please. Thanks!

**Comment by shivaditya-meduri** [comment]
> @shivaditya-meduri Any update on this PR? Please. Thanks!
I will submit a new PR for this issue, thank you all for your guidance.
